torch/optim/asgd.py

```diff
@@ -428,16 +428,7 @@ def _multi_tensor_asgd(
             torch._foreach_copy_(grouped_etas, new_etas)
         else:
             step = grouped_state_steps[0].item()
-            new_etas = []
-            new_mus = []
-
-            for i in range(len(grouped_mus)):
-                new_eta = _to_tensor(
-                    lr / (1 + lambd * lr * step**alpha), device=device
-                )
-                new_etas.append(new_eta)
-                new_mu = _to_tensor(1 / max(1, step - t0), device=device)
-                new_mus.append(new_mu)
-
-            torch._foreach_copy_(grouped_etas, new_etas)
-            torch._foreach_copy_(grouped_mus, new_mus)
+            new_eta = _to_tensor(lr / (1 + lambd * lr * step**alpha), device=device)
```

Comment:
btw, I think instead of custom `_to_tensor`could be used `torch.as_tensor` which can consume both python scalars and scalar tensors and a `device=` argument